0.14 Release

blakeblackshear opened the pull request:
Beta Docs: https://deploy-preview-11419--frigate-docs.netlify.app/

0.14 FAQ: https://github.com/blakeblackshear/frigate/discussions/12724

### Todo:
- [x] UI Rewrite
- [x] Authentication
  - [ ] look at ways to pass an authsig for notifications 
- [x] TLS

phedders commented:
Just saw this ticket and had to bump my install to the latest dev to see how it is going... and wow - the UI progress is amazing - Many thanks Nick and others doing all this incredible work. I was looking for stuff to 'criticize', or feedback but... not finding much yet...

remz1337 committed:
Add support for TensorRT v10 (multiple api calls have changed) (#11166)

* Add support for TensorRT v10 (multiple api calls have changed)

* Remove unnecessary size check in TensorRT v10 block

* Refactor to reduce code duplication

* Fix wrong function name in new _get_binding_dtype function and only return input check (not assertion) in new _binding_is_input function

* Add space around TRT_VERSION variable assignment (=) to respect linting

* More linting fix

* Update frigate/detectors/plugins/tensorrt.py

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

* More linting

---------

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

NickM-27 committed:
UI tweaks / fixes (#11479)

* Update coloring of logs button

* Fix paging after submitting

* Fix sorting

* Add icon

NickM-27 committed:
Update Deps (#11480)

* Update web deps

* Update python deps

* revert matplotlib

hawkeye217 committed:
Update beta docs (#11483)

* update beta docs

* fix relative coordinates

* fix relative coordinates

* changes

NickM-27 committed:
Fix current hour check for uneven timezones (#11484)

NickM-27 committed:
Adjust date math (#11485)

NickM-27 committed:
Fix extra space in command (#11487)

NickM-27 committed:
Safely load config file for go2rtc (#11491)

* Safely load config file for go2rtc

* Add log for default config

* Cleanup

hawkeye217 committed:
calculate aspect ratio of container to prevent overflow (#11495)

blakeblackshear committed:
master sync

blakeblackshear committed:
master sync v2

MarcA711 committed:
Update docs for rockchip platform (#11503)

* improve docs for rockchip

* update version info

* fix typo

* fix typo

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

* fix typo

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

---------

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

hawkeye217 committed:
ensure resizable handles are above jsmpeg canvas elements (#11519)

hawkeye217 committed:
bugfixes (#11526)

hawkeye217 committed:
redirect non existent paths to live view (#11536)

NickM-27 committed:
Tweaks and fixes (#11541)

* Update config version to be stored inside of the config

* Don't remove items from list when navigating back

* Use video api instead of webps for live current hour filmstrip

* Check that the config file is writable

* Show camera name when camera is offline

* Show camera name when offline

* Cleanup

hawkeye217 committed:
save video dimensions in onLoadedData instead of onLoadedMetadata (#11545)

NickM-27 committed:
Fix optimistic state (#11556)

hawkeye217 committed:
docs update: how to exclude camera from alerts/detections (#11560)

hawkeye217 committed:
refresh editor value when config is updated (#11559)

NickM-27 committed:
Fix live view updating when it shouldn't be (#11561)

* Simplify live image update logic

* Fix case where go2rtc is not setup

blakeblackshear committed:
upgrade to latest openvino version (#11563)

NickM-27 committed:
Add ability to interact with review items in events list (#11562)

* Add ability to interact with review items

* Ignore on iOS

* Don't load metadata

* Bug fixes

hawkeye217 committed:
Make jsmpeg players fully responsive (#11567)

* make jsmpeg canvas responsive

* make birdseye responsive too

NickM-27 committed:
remove libusb build (#11571)

NickM-27 committed:
Fixes (#11575)

* Fix settings icon

* Handle out of resources

NickM-27 committed:
Handle case where preview doesn't automatically changeover (#11583)

hawkeye217 committed:
Live view tweaks and jsmpeg bugfix (#11584)

* live view tweaks and jsmpeg bugfix

* use container aspect in check

hawkeye217 committed:
clean up unneeded code (#11587)

NickM-27 committed:
Simplify timezone math (#11586)

* Use utc minutes

* Cleanup

hawkeye217 committed:
check if motion dataframe is empty (#11593)

NickM-27 committed:
UI fixes (#11596)

* Fix using undefined search params

* Fix calendar selection

* Simplify

hawkeye217 committed:
ensure the correct container is used for canvas calcs (#11599)

jamie-tm committed:
Draw motion mask first (#11598)

This draws the motion mask before the other overlay elements (such as bounding boxes), so that they are still visible. Fixes #11592

NickM-27 committed:
Fix iOS fullscreen (#11600)

timgurske commented:
I have been running this version for several days now and while it's not as stable as 0.13 (web socket errors in Firefox) I think it's a **huge improvement**. Thank you for working on this!

NickM-27 committed:
UI Fixes (#11602)

* Fix playback rate not showing

* Fix export image

* Formatting

* Formatting

NickM-27 committed:
Always use mse unless webrtc is requrested (#11605)

NickM-27 committed:
cache current hour previews (#11606)

NickM-27 committed:
Cache camera stream info to speed up future config generations (#11614)

* Cache camera stream info to speed up future config generations

* Formatting

* fix

NickM-27 committed:
Settings rework (#11613)

* refactor settings to be consistent with other page structure

* Implement non auto live

* Adjust missing view

* Quick fix

* Clarify settings options

Co-authored-by: Josh Hawkins <32435876+hawkeye217@users.noreply.github.com>

* Update naming and config restarts

* Rename

---------

Co-authored-by: Josh Hawkins <32435876+hawkeye217@users.noreply.github.com>

hawkeye217 committed:
string or list for camera groups (#11616)

hawkeye217 committed:
Various bugfixes and improvements (#11624)

* various bugfixes and improvements

* add separator

* no separator

NickM-27 committed:
Improve efficiency of log and metrics pages (#11622)

* Rework stats pages

* Handle limited data case

* Handle page and arrow keys

* Adjust sizing

NickM-27 committed:
Remove live mode from config (#11618)

* Use preferred mode as default

* Remove live mode from config

* Add deer icon

* remove from config schema

hawkeye217 committed:
fix birdseye fullscreen (#11625)

NickM-27 committed:
Always save previews in UTC offset (#11629)

hawkeye217 committed:
assume 1 grid column by default on live dashboard (#11630)

NickM-27 committed:
Don't show mark reviewed button when all items are in progress (#11636)

* Don't show mark reviewed button when all items are in progress

* Fix unknown preview file

hawkeye217 committed:
static handlebar size to better match figma (#11638)

NickM-27 committed:
Handle case where user sets detections as empty list (#11646)

hawkeye217 committed:
non-modal dropdown menus (#11649)

NickM-27 committed:
UI Fixes (#11648)

* Add cursor pointer to preset dropdown

* Catch key index

* Fix iOS mime type

NickM-27 committed:
Update dialog sizing for plus dialog (#11650)

hawkeye217 committed:
don't sleep mobile devices when fullscreen (#11652)

NickM-27 committed:
Adjust nginx proc count based on available CPUs (#11653)

* Restrict nginx to 4 processes if more are available

* Fix bash

* Different sed structure

* Limit ffmpeg thread counts for secondary ffmpeg processes

* Add up / down keyboard shortcut

track0x1 committed:
Fix Statusbar rendering NaN% for unsupported GPUs (#11655)

hawkeye217 committed:
fix race where camera change effect sometimes was called after layout building (#11656)

hawkeye217 committed:
stay in fullscreen when navigating to a camera (#11666)

NickM-27 committed:
Handle error when live view stalls (#11665)

* Handle error when live view stalls

* Manually calculate buffer timeout

* Formatting

NickM-27 committed:
Don't fail if user has bind mounted nginx config (#11671)

blakeblackshear committed:
Update deps (#11679)

* update web deps

* update actions

* automatic stable tag publishing

* python deps

* typo

sorenisanerd committed:
feat: apply ingress path to app paths (#11677)

When serving Frigate at a subpath, the paths that show in the URL bar
and that wind up in your browser history are anchored at the web root.
I.e. you go to `https://example.com/frigate/`, it changes to
`https://example.com/`, and clicking around works as expected, but the
`frigate/` prefix is gone.

It's confusing if you don't know that the URL's are entirely virtual.
Also, your browser history is useless, since the URL's point to e.g.
`https://example.com/#kitchen`, but visiting that URL will not hit
`/frigate/` at all.

Most of the work is already done. Nginx injects javascript to set
`window.baseURL` based on the X-Ingress-Path header. This change passes
that to BrowserRouter, so that it'll be part of the URL's it shows.

Fixes #4526

NickM-27 committed:
Fix path containing too many / (#11680)

blakeblackshear committed:
TLS support (#11678)

* implement self signed cert and monitor/reload

* move go2rtc upstream to separate file

* add directory for ACME challenges

* make certsync more resilient

* add TLS docs

* add jwt secret info to docs

hawkeye217 committed:
container for birdseye aspect and auto width for mobile time pill (#11685)

blakeblackshear committed:
improve tls implementation (#11690)

* improve tls implementation

* update docs

blakeblackshear committed:
fix tempio install for arm64 (#11691)

hawkeye217 committed:
UI tweaks and bugfixes (#11692)

* UI tweaks and bugfixes

* fix linter complaints in unmodified files

MarcA711 committed:
Add coco-80 labelmap and update FFmpeg for Rockchip  (#11695)

* add coco-80 labelmap and update ffmpeg

* Update docs/docs/configuration/object_detectors.md

---------

Co-authored-by: Blake Blackshear <blake.blackshear@gmail.com>

hawkeye217 committed:
Use valid/unique css identifier for jsmpeg canvas elements (#11704)

NickM-27 committed:
Add bug template to issues (#11712)

* Add bug template to issues

* Delete .github/ISSUE_TEMPLATE/bug_report.yml

* Create bug-report.yml

* Create bug-link.yml

NickM-27 committed:
Fix bug template (#11716)

* Update bug-link.yml

* Delete .github/ISSUE_TEMPLATE/bug-link.yml

* Use existing template

NickM-27 committed:
fix duplicate id in bug report (#11718)

hawkeye217 committed:
Ensure debug switches match loaded options (#11721)

reidprichard committed:
Update authentication.md to note port 8080 vs 5000 (#11722)

* Update authentication.md to note port 8080 vs 5000

* Update docs/docs/configuration/authentication.md

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

---------

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

hawkeye217 committed:
Theme scrollbars with tailwind-scrollbar (#11723)

NickM-27 committed:
Ensure review padding is consistently applied (#11728)

hunterjm committed:
Fix unclean shutdown of ZMQ (#11740)

NickM-27 committed:
UI Fixes (#11742)

* Allow deleting failed in progress exports

* Fix comparison and preview retrieval

* Fix stretching of event cards

* Reset edit state when group changes

* Allow specifying group

hawkeye217 committed:
Try webrtc when mse fails with decoding error (#11745)

* Try webrtc if enabled and mse fails with decoding error

* default to jsmpeg if webrtc times out

* check for mic first

NickM-27 committed:
Don't show preview for birdseye (#11749)

* Don't show preview for birdseye

* Retry ws connection on error

* Flex wrap cameras labels

ramunasd committed:
fix: remove contradictory ffmpeg param and missing param (#11752)

hawkeye217 committed:
Some small layout tweaks for portrait cams and motion review (#11766)

* Some small layout tweaks for portrait cams and motion review

* spans

* fix desktop

felipecrs committed:
Ensure nginx worker processes respects docker limits (#11769)

* Ensure nginx worker processes respects docker limits

* Update get_cpus.sh revision

* Add get_cpus.sh functionality inline to nginx/run

hawkeye217 committed:
Draggable grid layout bugfixes (#11777)

* Maintain aspect ratio when overdragging

* add existing x value

* Better handle portrait and wide cam aspect ratios

hawkeye217 committed:
Change debug message about deleting db entries to warning (#11780)

hawkeye217 committed:
Ensure export thumbnail datetime is UTC (#11786)

hawkeye217 committed:
Ensure datetime comparison is the same (native vs aware) (#11789)

hawkeye217 committed:
Compare timestamps instead of datetimes when exporting (#11790)

blakeblackshear committed:
Shutdown hang (#11793)

* intentionally handle queues during shutdown and carefully manage shutdown order

* more carefully manage shutdown to avoid threadlocks

* use debug for signal logging

* ensure disabled cameras dont break shutdown

* typo

blakeblackshear committed:
Update deps (#11799)

* web deps

* python deps

blakeblackshear committed:
Adds support for YOLO-NAS in OpenVino (#11645)

* update onnxruntime

* support for yolo-nas in openvino

* cleanup notebook

* update docs

* improve docs

* handle AUTO issue and update docs

hawkeye217 committed:
UI tweaks (#11795)

* Prevent "undefined" from being displayed in searchParams string

* Show message for no motion data

* Use theme colors for no preview found divs

hawkeye217 committed:
Retain 3 columns on desktop motion review (#11805)

hawkeye217 committed:
Change breakpoint for desktop motion review columns (#11808)

hawkeye217 committed:
Make sure camera is always set in settings (#11812)

NickM-27 committed:
Remove mention of recordings timeline object debugging (#11820)

blakeblackshear committed:
update images in readme

hawkeye217 committed:
Prevent editing of object mask type on existing mask (#11829)

blakeblackshear committed:
move clip.mp4 backend to clips folder (#11834)

* move clip.mp4 backend to clips folder

* improve caching

* fix check

hawkeye217 committed:
Live player fixes and improvements (#11855)

* Only set stalled error when player is visible

* Show activity indicator before live player starts playing

* remove comment

* keep gradients when still image is showing

* fix chips

* red dot and outline

NickM-27 committed:
UI changes (#11863)

* Delay live ready being dropped

* Handle case of removed camera

blakeblackshear committed:
just check for secret file specifically (#11877)

* just check for secret file specifically

* add josh to funding

NickM-27 committed:
Review improvements (#11879)

* Update segment even when number of active objects is the same

* add score to frigate+ chip

* Add support for selecting zones

* Add api support for filtering on zones

* Adjust UI

* Update filtering logic

* Clean up

hawkeye217 committed:
Review UI improvements (#11882)

hawkeye217 committed:
UI fixes (#11883)

* Keep as optional prop

* put zones inside of scrollable container

hawkeye217 committed:
Add scrollbar class to preview row/column (#11890)

NickM-27 committed:
Copy review data so there is a diff (#11896)

felipecrs committed:
Fix cpu count when process name includes the word processor (#11911)

hawkeye217 committed:
Activity indicator whenever preferred live mode changes (#11913)

hawkeye217 committed:
Activity indicator for alerts/detections count when loading (#11914)

* Activity indicator for alerts/detections count when loading

* Return zeros if summary is unavailable

hawkeye217 committed:
Ensure uri components are decoded (#11920)

NickM-27 committed:
Correctly update segment data (#11922)

NickM-27 committed:
UI Tweaks (#11931)

* Show number of items instead of dot

* Don't call error when connection has been closed on purpose

* Use motion icon for motion

* Show text on tablets as well

hawkeye217 committed:
Draggable grid fixes (#11944)

* Use globals on grid for resizing/dragging flags

* remove unneeded useeffect

NickM-27 committed:
UI tweaks (#11940)

* Enforce events must have snapshots for frigate+

* Open docs links in separate tabs

* Reload after restart to the baseUrl

blakeblackshear committed:
clarifications for proxy auth mode (#11947)

NickM-27 committed:
Manually set current time when selecting event (#11948)

* Manually set current time when selecting event

* Make it clear which camera has no preview

* Make it clear which camera has no preview

* Format camera name

blakeblackshear committed:
clarify plus models (#11951)

blakeblackshear committed:
clarify model id (#11953)

blakeblackshear committed:
Merge remote-tracking branch 'origin/master' into dev

MarcA711 committed:
Update FFmpeg for Rockchip (#11952)

NickM-27 committed:
Fix calendar selection (#11959)

blakeblackshear committed:
split out proxy from auth (#11963)

* split out proxy from auth

* update documentation

* fixup auth mode check

alexyao2015 committed:
Show failure exception message (#11964)

blakeblackshear committed:
set shortest edge to preview height (#11971)

hawkeye217 committed:
Optional proxy in config (#11973)

hawkeye217 committed:
Consolidate onvif camera recommendations in docs (#11972)

NickM-27 committed:
Update template to include all detector types (#11981)

* Change from coral to general detector dropdown

* Update camera-support.yml

* Update config-support.yml

* Update general-support.yml

miguelangel-nubla committed:
Fix header auth (#11985)

miguelangel-nubla committed:
Fix X-Proxy-Secret header passthrough (#11984)

blakeblackshear committed:
enable tls by default if undefined (#11994)

hawkeye217 committed:
Use modal on dropdowns for mobile only (#11993)

NickM-27 committed:
UI Tweaks (#12002)

* Adjust review padding

* Fix mse check

* Don't fail when cpu property is missing

* ignore lines without any spaces

NickM-27 committed:
Don't fail when preview restore fails (#12022)

* Don't fail when preview restore fails

* Cleanup

NickM-27 committed:
Set image height to make bandwidth usage lower (#12024)

NickM-27 committed:
Increase resolution for mobile viewing (#12025)

NickM-27 committed:
Use api path to determine type (#12031)

* Use api path to determine type

* Use in both cases

* Fix extension parsing

hawkeye217 committed:
Small jsmpeg and mse player fixes (#12032)

hawkeye217 committed:
Memoize onPlaying and only instantiate one jsmpeg player (#12033)

NickM-27 committed:
Fix case where S6 timestamp is missing (#12040)

MarcA711 committed:
fix key error for custom models (#12042)

NickM-27 committed:
Add ability to select all on desktop (#12044)

* Add ability to select all review items

* Refactor keybaord listener

NickM-27 committed:
Treat meta key same as ctrl (#12051)

hawkeye217 committed:
Various UI tweaks (#12061)

Samywamy10 committed:
Update web readme (#12062)

* Update web readme

* Update /web readme

* Apply suggestions from code review

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

---------

Co-authored-by: Sam Wright <sam@sams-mbp.lan>
Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

NickM-27 committed:
Update go2rtc (#12063)

NickM-27 committed:
Fix showing loading indicator when first loading a camera without previews (#12064)

NickM-27 committed:
fix case where camera is disabled and has no previews (#12066)

* fix case where camera is disabled and has no previews

* Maintain slow loading behavior

hawkeye217 committed:
Timeline performance improvements (#12070)

* Use intersection observer for timeline segments

* only render when visible

hawkeye217 committed:
Only close MSE websocket when it's already open (#12078)

hawkeye217 committed:
Prevent ptz keyboard shortcuts from reopening presets menu (#12079)

NickM-27 committed:
Fix export timing (#12080)

NickM-27 committed:
Auto focus user field for login (#12083)

NickM-27 committed:
Simplify zone objects example (#12086)

NickM-27 committed:
Fix preview failing on cameras with - in name (#12093)

NickM-27 committed:
Add reviews topic to MQTT docs (#12097)

NickM-27 committed:
Fix left swipe on preview (#12104)

* Fix left swipe

* Simplify

NickM-27 committed:
Show correct previous state when updating for end (#12122)

* Show correct previous state when updating for end

* remove log

* Formatting

NickM-27 committed:
Fix review items (#12126)

NickM-27 committed:
Downgrade go2rtc (#12139)

OnFreund committed:
Update TLS docs with certbot instructions (#12141)

* Update tls.md

Update TLS docs with certbot instructions

* Apply suggestions from code review

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

---------

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

NickM-27 committed:
Quick fix (#12153)

* Use list for zones to keep chronological order

* Replace when changing playbackRate

NickM-27 committed:
Nginx config tweaks (#12174)

* Change auth port and remove ipv6

* Add docs for nginx bind mount

* Consolidate listen statements

* Update port in docs

* Fix typing

NickM-27 committed:
Fix yamnet model download (#12200)

hawkeye217 committed:
Live player fixes and improvements (#12202)

* Live player fixes and improvements

* remove comment

* Simplify wording

NickM-27 committed:
Fix audio model download again (#12207)

* Fix audio model download again

* Update Dockerfile

NickM-27 committed:
Fix nginx 5000 template (#12210)

hawkeye217 committed:
Fix jsmpeg player flickering (#12213)

hawkeye217 committed:
Check websocket readyState for disconnect and fix firefox pip (#12216)

hawkeye217 committed:
Use camera name instead of stream_name for jsmpeg players (#12219)

hawkeye217 committed:
Use cache key for mask/zone editor image (#12232)

hawkeye217 committed:
Re-center ptz controls in mobile landscape and prevent text selection (#12242)

NickM-27 committed:
Use latest 5.1 ffmpeg update (#12243)

* Use latesat 5.1 ffmpeg update

* Fix arm build

NickM-27 committed:
Change qsv device arg to standard hwaccel arg (#12249)

hawkeye217 committed:
Add fullscreen button to the default live grid on desktops (#12250)

NickM-27 committed:
Update ffmpeg source (#12251)

* Revert "Use latest 5.1 ffmpeg update (#12243)"

This reverts commit 93e08688be9b37d9a605f8606c92d2aa34aae7c8.

* Revert "Change qsv device arg to standard hwaccel arg (#12249)"

This reverts commit 56b4a551dc35698f12035d290ab0b1de275bb920.

* Use different repo for build

NickM-27 committed:
Enforce minimum value for mqtt stats update (#12253)

SConaway committed:
docs(go2rtc): troubleshooting improvements (#12192)

hawkeye217 committed:
Ensure MSE onPlaying always gets called, even if loadeddata never fires (#12271)

* Ensure MSE onPlaying always gets called, even if loadeddata never fires

* Call handleLoadedMetadata too if not playing yet

hawkeye217 committed:
Apply landscape margin to ptz controls on mobile only (#12272)

hawkeye217 committed:
Add exports message and default to webrtc on < iOS 17.1 (#12281)

hawkeye217 committed:
Update export docs (#12288)

* Update exporting docs

* add timelapse clarification

hawkeye217 committed:
Remove matplotlib and generate color palette to mimic matplotlib's colors (#12327)

ubawurinna committed:
OpenVino - Error clarification for compile_model function (#12304)

* Error clarification for openvino's compile_model function

* run ruff format

---------

Co-authored-by: ubawurinna <you@example.com>

hawkeye217 committed:
UI tweaks (#12297)

* Use full resolution aspect for main camera style in history view

* Only check for offline cameras after 60s of uptime

* only call onPlaying when loadeddata is fired or after timeout

* revert to inline funcs

* Portal frigate plus alert dialog

* remove duplicated logic

* increase onplaying timeout

* Use a ref instead of a state and clear timeout in AutoUpdatingCameraImage

* default to the selected month for selectedDay

* Use buffered time instead of timeout

* Use default cursor when not editing polygons

NickM-27 committed:
Limited shm frame count (#12346)

* Only keep 2x detect fps frames in SHM

* Don't delete previous shm frames in output

* Catch case where images do not exist

* Ensure files are closed

* Clear out all frames when shutting down

* Correct the number of frames saved

* Simplify empty shm error handling

* Improve frame safety

NickM-27 committed:
Revert "Limited shm frame count (#12346)" (#12362)

This reverts commit 34812b74399c569680a48c60d25f6d22a84b80c8.

stephendb committed:
Bug fix for ONVIF cameras, adjust_time parameter added (#12352)

* adds adjust_time which allows users to fix an issue with onvif authentication where time is not syncrhonized

* updated adjust_time to ignore_time_mismatch to make it clearer what this option does

* adds ignore_time_mismatch to the reference.md and adds a line about the security risk this can introduce as well as the recommendation to setup NTP for both ends.

* fix format error

* happy now linter?

* white space

* Update docs/docs/configuration/reference.md

Co-authored-by: Josh Hawkins <32435876+hawkeye217@users.noreply.github.com>

---------

Co-authored-by: Stephen Butler <stephen.butler@ni.com>
Co-authored-by: Josh Hawkins <32435876+hawkeye217@users.noreply.github.com>

hawkeye217 committed:
Various UI tweaks and changes (#12364)

hawkeye217 committed:
Theme updates (#12373)

* remove hideous and ugly themes

* incorporate dei into ui design

* neutral as a theme color

* high contrast theme adjustments

* color tweaks

NickM-27 committed:
Docs tweaks (#12375)

* Add explanation of bounding box colors

* Add FAQ about offline camera

hawkeye217 committed:
Clickable logo on desktop sidebar and useMatch for camera group visibility (#12379)

NickM-27 committed:
Fix review update causing api spam (#12387)

NickM-27 committed:
Update Web deps (#12388)

* Update radix ui

* Update vite

* More ui deps

* Update typscript

* Update react router

hawkeye217 committed:
Fix linter warnings on color order (#12389)

hawkeye217 committed:
Fix debug camera image not updating when loading (#12394)

NickM-27 committed:
Simplify ws updating (#12390)

* Simplify ws updating

* Simplify return values

ThorpeJosh committed:
strip whitespaces when loading secrets (#12393)

* strip whitespaces when loading secrets

* formatting

NickM-27 committed:
Update review docs (#12401)

hawkeye217 committed:
Move review classification settings to camera settings view (#12410)

* Camera settings view for alerts/detections

* flxes, beautifying, zone renaming, clean up

* replace underscores with spaces in zone names

* replace underscores with spaces in labels

hawkeye217 committed:
Display activity indicators when debug and mask/zone images load (#12411)

hawkeye217 committed:
Ensure detections are cleared when limit box is unchecked (#12412)

hawkeye217 committed:
Attempt to support zoom-only onvif cams (#12415)

* Attempt to support zoom-only onvif cams

* don't skip ptz configuration

NickM-27 committed:
Recordings improvements (#12417)

NickM-27 committed:
Don't detach media (#12421)

NickM-27 committed:
Improve safari image scrolling performance (#12429)

* Don't set z-height on iOS

* More z-index cleanup

blakeblackshear committed:
Merge remote-tracking branch 'origin/master' into dev

hawkeye217 committed:
Update docs for clarity on review items (#12441)

hawkeye217 committed:
No need to check for h264 onvif profile (#12444)

NickM-27 committed:
Filter alerts by camera group (#12456)

NickM-27 committed:
Fix camera filtering logic (#12457)

* Fix camera filtering logic

* Cleanup

* Simplify and consider birdseye only group in logic

* Don't add filter when group is birdseye only

hawkeye217 committed:
Fix flashing of previous still image when live player stops (#12458)

NickM-27 committed:
Fix handling of default group (#12459)

hawkeye217 committed:
Check if camera is active before disabling liveReady (#12461)

hawkeye217 committed:
Use refs for proper js closures in the liveReady timeout (#12464)

NickM-27 committed:
Fix ui config migration (#12476)

NickM-27 committed:
Don't allow backwards recordings (#12477)

NickM-27 committed:
Add camera name to recordings log (#12480)

* Add camera name to recordings log

* Formatting

hawkeye217 committed:
Prevent onPlaying from being called repeatedly in jsmpeg player (#12482)

NickM-27 committed:
UI fixes (#12490)

* Improve export handling when errors occur

* Fix mobile zooming

* Handle recordings buffering

* Cleanup

* Url encode export name

* Start with actual name in input

* Fix buffering

hawkeye217 committed:
User-selectable weekday start (Sunday/Monday) for review calendar (#12491)

hawkeye217 committed:
Use canvas2d renderer for jsmpeg player (#12498)

hawkeye217 committed:
Reset preferred live modes to defaults on window visibility change (#12499)

hawkeye217 committed:
Disable web assembly for jsmpeg player (#12502)

NickM-27 committed:
Add more icons to event icon types (#12507)

NickM-27 committed:
Fix plus sorting button (#12513)

NickM-27 committed:
UI fixes (#12542)

* Don't require previews to show motion ui

* Fix recording text to match hls player logic

NickM-27 committed:
Persist show reviewed locally so it maintains state (#12560)

* Persist show reviewed locally so it maintains state

* fix

* Theming fixes

NickM-27 committed:
UI Tweaks (#12571)

NickM-27 committed:
Use JSON instead of pickle for serialization (#12590)

hawkeye217 committed:
Prevent pandas overflow and runtime errors from division by zero/NaN (#12591)

* Prevent pandas overflow and runtime errors from division by zero/NaN

* remove pysqlite3

NickM-27 committed:
Use pickle for config objects (#12594)

NickM-27 committed:
Ensure that persisted state is kept in sync (#12596)

NickM-27 committed:
Fix perview serialization (#12597)

hawkeye217 committed:
loitering_time can be zero (#12599)

NickM-27 committed:
Docs updates (#12607)

hawkeye217 committed:
Add filmstrip video/image toggle to general settings (#12608)

NickM-27 committed:
Fix frigate failing when no config is defined (#12611)

NickM-27 committed:
Limit preview threads (#12633)

NickM-27 committed:
Show skeleton when live filmstrip items are loading (#12660)

NickM-27 committed:
apply iOS fix to safari (#12663)

NickM-27 committed:
Build libusb for coral compatibility (#12681)

hawkeye217 committed:
Display messages when no events exist (#12694)

* Display message in desktop events list when no events exist

* Add message for when no events are found on plus view

* validating check

* activity indicator check

* clarify error message

NickM-27 committed:
Remove mention of older yolox model (#12711)

NickM-27 committed:
Update docs (#12714)

NickM-27 committed:
Fix tall videos from covering height in export page (#12725)

* Fix tall videos from covering height in export page

* Handle mobile landscape

kerneljake committed:
Add example of single pcie coral (#12446)

* Update object_detectors.md

* Update docs/docs/configuration/object_detectors.md

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

---------

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

NickM-27 committed:
Installation and getting started docs improvements (#12395)

* Add tip banner for quick links

* Add link to getting started guide from installation

* Remove dummy config

* Move tip

* Clarify installation docs

Co-authored-by: Blake Blackshear <blake@frigate.video>

* Update docs/docs/guides/getting_started.md

Co-authored-by: Blake Blackshear <blake@frigate.video>

---------

Co-authored-by: Blake Blackshear <blake@frigate.video>

blakeblackshear committed:
always release from dev builds
